ethereum_history_api/circuits/lib/src/misc/arrays_test.nr

```diff
@@ -1,97 +1,118 @@
-use crate::misc::arrays::{sub_array_equals_up_to_length, sub_array_equals, array_equals, resize};
+use crate::misc::arrays::memcpy;
 
 #[test]
-fn test_assert_sub_array_equals() {
-    let value: [Field; 4] = [1, 2, 0, 0];
-    let array: [Field; 6] = [42, 1, 2, 3, 4, 5];
-    assert(sub_array_equals_up_to_length(value, array, 1, 2));
+fn test_memcpy() {
+    let mut dest = [0; 3];
+    let src = [1, 2, 3, 4, 5];
+    memcpy(&mut dest, src, 2);
+    assert(dest == [3, 4, 5]);
 }
```

Comment:
Maybe some failing test (Index array out of bound) for `memcpy` with cases:
* offset >= M
* offset + N >= M